pkgs/games/gdlauncher/default.nix

```diff
@@ -0,0 +1,30 @@
+{ pkgs, lib, ... }:
+let
+  pname = "gdlauncher";
+  version = "1.1.30";
+  name = "${pname}-${version}";
+  src = pkgs.fetchurl {
+    url = "https://github.com/gorilla-devs/GDLauncher/releases/download/v${version}/GDLauncher-linux-setup.AppImage";
+    hash = "sha256-4cXT3exhoMAK6gW3Cpx1L7cm9Xm0FK912gGcRyLYPwM=";
+  };
+
+  appimageContents = pkgs.appimageTools.extractType2 { inherit name src; };
+in
+pkgs.appimageTools.wrapType2 rec {
```

Comment:
the rec keyword here stands for recursive, if it's set (which it doesn't have to be) you can use attributes defined in the same scope in which say are defined.